packages/contracts-bedrock/test/safe-tools/SafeTestTools.sol

```diff
@@ -0,0 +1,509 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.0 <0.9.0;
+
+import "forge-std/Test.sol";
+import "scripts/libraries/LibSort.sol";
+import { ModuleManager, GuardManager, Safe as GnosisSafe } from "safe-contracts/Safe.sol";
+import { SafeProxyFactory as GnosisSafeProxyFactory } from "safe-contracts/proxies/SafeProxyFactory.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { SignMessageLib } from "safe-contracts/libraries/SignMessageLib.sol";
+import "./CompatibilityFallbackHandler_1_3_0.sol";
+
+// Tools to simplify testing Safe contracts
+// Author: Colin Nielsen (https://github.com/colinnielsen/safe-tools)
+
+address constant VM_ADDR = 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D;
+bytes12 constant ADDR_MASK = 0xffffffffffffffffffffffff;
+
+/// @dev Get the address from a private key
+function getAddr(uint256 pk) pure returns (address) {
+    return Vm(VM_ADDR).addr(pk);
+}
+
+/// @dev Get arrays of addresses and private keys. The arrays are sorted by address, and the addresses are labelled
+function makeAddrsAndKeys(uint256 num) returns (address[] memory addrs, uint256[] memory keys) {
+    keys = new uint256[](num);
+    addrs = new address[](num);
+    for (uint256 i; i < num; i++) {
+        uint256 key = uint256(keccak256(abi.encodePacked(i)));
+        keys[i] = key;
+    }
+
+    for (uint256 i; i < num; i++) {
+        addrs[i] = Vm(VM_ADDR).addr(keys[i]);
+        Vm(VM_ADDR).label(getAddr(keys[i]), string.concat("SAFETEST: Signer ", string(abi.encodePacked(bytes32(i)))));
+    }
+}
```

Comment:
`getAddr` was already in this file and is used below. Although it is just a wrapper I'd like to keep it for now to avoid the refactor it'd take to remove it.

`makeAddrsAndKeys` could no longer be  a free function if it was dependendent on `makeAddrAndKey`. 
Replacing all invocations of `makeAddrsAndKeys` with a loop would be somewhat ugly because you'd need to manually initialize two arrays, and add the values to them one at a time, and then label the addresses.

I took a stab at pulling all the free functions in to the `SafeTestTools` contract, but there are also deps on them in the `SafeTestLib`. Unfortunately I think we're going to have to inherit some warts from this code, but IMO the benefits are worthwhile.

Reply:
> `getAddr` was already in this file and is used below. Although it is just a wrapper I'd like to keep it for now to avoid the refactor it'd take to remove it.

Sounds good 👌

> `makeAddrsAndKeys` could no longer be  a free function if it was dependendent on `makeAddrAndKey`. 
> Replacing all invocations of `makeAddrsAndKeys` with a loop would be somewhat ugly because you'd need to manually initialize two arrays, and add the values to them one at a time, and then label the addresses.

Ah sorry I wasn't clear, I meant keep the `makeAddrsAndKeys` but replace it's implementation with calls to `makeAddrsAndKey`

> I took a stab at pulling all the free functions in to the `SafeTestTools` contract, but there are also deps on them in the `SafeTestLib`. Unfortunately I think we're going to have to inherit some warts from this code, but IMO the benefits are worthwhile.

Leaving it seems fine overall. Maybe something worth tracking in the tech debt task...

Reply:
> Ah sorry I wasn't clear, I meant keep the makeAddrsAndKeys but replace it's implementation with calls to makeAddrsAndKey

Yes, I tried that, but I don't think it's possible because free functions can't use inherited code.